### PR TITLE
Reduce splat VRAM size

### PR DESCRIPTION
**Pull request by CedricGuillemet**
From 64 bytes per splat to 32 (36 for webgl1).

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15720/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15720/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15720/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15720/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15720/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15720/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15720/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15720/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15720/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15720/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15720/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

File: packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts [hunk 1]
```diff
@@ -95,6 +96,8 @@ export class GaussianSplattingMesh extends Mesh {
         new SubMesh(0, 0, 4, 0, 6, this);
 
         this.setEnabled(false);
+        // webGL2 and webGPU support for RG texture with float16 is fine. not webGL1
+        this._useRGBACovariants = !this.getEngine().isWebGPU && this.getEngine().version === 1.0;
```
Comment (Popov72) [inline packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts:100]: You don't have to support WebGL1, see @deltakosh answer offline!

File: packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts [hunk 2]
```diff
@@ -505,12 +512,71 @@ export class GaussianSplattingMesh extends Mesh {
...
+            let factor = -10000;
+            for (let covIndex = 0; covIndex < 6; covIndex++) {
+                factor = Math.max(factor, Math.abs(covariances[covIndex]));
+            }
+
+            this._splatPositions[4 * i + 3] = factor;
+            const transform = factor;
+
+            // overflow checks removed for best performance
+            const toFloat16 = function (val: number) {
```
Comment (Popov72) [inline packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts:532]: We have this function in `Misc/textureTools.ts` (`ToHalfFloat`).

File: packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts (hunk 1 above)
Comment (CedricGuillemet) [reply packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts:101]: without this support, checks will not pass.

**Review by sebavan — APPROVED** [review]

**Pull request merged**